flink-clients/src/test/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrapTest.java

```diff
@@ -265,6 +265,35 @@ public void testApplicationFailsAsSoonAsOneJobFails() throws Throwable {
         assertEquals(exception.getStatus(), ApplicationStatus.FAILED);
     }
 
+    @Test
+    public void testApplicationFailsIfJobIsAlreadyMarkedAsClean() throws Throwable {
+        final Configuration configuration = getConfiguration();
+        final JobID jobId = new JobID();
+        configuration.set(PipelineOptionsInternal.PIPELINE_FIXED_JOB_ID, jobId.toHexString());
+
+        final ApplicationDispatcherBootstrap bootstrap =
+                new ApplicationDispatcherBootstrap(
+                        getProgram(1),
+                        Collections.emptyList(),
+                        actualJobId -> true,
+                        configuration,
+                        // passing null should not lead to any problem because we want to fail
+                        // before doing any call to the dispatcher
+                        null,
```

Comment:
eh. Why should this test fail if I eagerly retrieve the address from the dispatcher gateway?